internal/core/item_rendering.rs

```diff
@@ -930,14 +969,14 @@ impl<'a, T: ItemRenderer + ItemRendererFeatures> ItemRenderer for PartialRendere
                     .get_or_insert_with(|| Box::pin(PropertyTracker::default()))
                     .as_ref()
                     .evaluate_if_dirty(|| *data = eval());
-                data.geometry
+                *data.geometry()
             }
             None => {
                 let cache_entry = crate::graphics::CachedGraphicsData::new(eval);
                 let geom = cache_entry.data.clone();
                 rendering_data.cache_index.set(cache.insert(cache_entry));
                 rendering_data.cache_generation.set(cache.generation());
-                geom.geometry
+                *geom.geometry()
```

Comment:
This function here, `filter_item` in the `PartialRenderer` is the reason why `CachedItemGeometryAndTransform` needs to store not only the translation but the entire geometry of an item.

As-is in this patch, `CachedItemGeometryAndTransform` uses 48 bytes on aarch64. If I change this code to fetch the geometry from the item directly, then I can reduce the geometry in `CachedItemGeometryAndTransform` to just a translation and the size goes to 40 bytes.

Reply:
I see.
I think it would be better to re-query the geometry again (untracked).
The reason filter_item returns the geometry is because it had the geometry for free. But IMHO it is better to query it again (untracked) than keeping it in the cache.

We could optimize for the common case of
 - Geometry == bounding rect
 - bounding rect is empty